### ISSUE TITLE
Enhance a11y for `DatePicker`

### ISSUE THREAD

**Issue by hirotomoyamada** [yamada-ui/yamada-ui#1251]
path: stories/components/forms/date-picker.stories.tsx

### Description
2 accessibility violations were found in the `DatePicker`.

#### [withDefaultValue](https://yamada-ui.github.io/yamada-ui/?path=/story/components-forms-datepicker--with-default-value)

Expected the HTML found at $('input') to have no violations:
    <input class="ui-date-picker__field__input css-hm2sw1" tabindex="0" value="2024/04/21">

    Received:
    Form elements must have labels (label)

    Fix any of the following:
      Form element does not have an implicit (wrapped) <label>
      Form element does not have an explicit <label>
      aria-label attribute does not exist or is empty
      aria-labelledby attribute does not exist, references elements that do not exist or references elements that are empty
      Element has no title attribute
      Element has no placeholder attribute
      Element's default semantics were not overridden with role="none" or role="presentation"

    You can find more information on this issue here:
    https://dequeuniversity.com/rules/axe/4.9/label?application=axeAPI

#### [withLocale](https://yamada-ui.github.io/yamada-ui/?path=/story/components-forms-datepicker--with-locale)

Expected the HTML found at $('input') to have no violations:
    <input class="ui-date-picker__field__input css-hm2sw1" tabindex="0" value="">

    Received:
    Form elements must have labels (label)

    Fix any of the following:
      Form element does not have an implicit (wrapped) <label>
      Form element does not have an explicit <label>
      aria-label attribute does not exist or is empty
      aria-labelledby attribute does not exist, references elements that do not exist or references elements that are empty
      Element has no title attribute
      Element has no placeholder attribute
      Element's default semantics were not overridden with role="none" or role="presentation"

    You can find more information on this issue here:
    https://dequeuniversity.com/rules/axe/4.9/label?application=axeAPI

### Problem Statement/Justification
A11y is essential to ensure quality when operating a package.

### Additional Information
While it's ideal to cover all jobs, if there are too many jobs to work on, please create a PR within your capacity.
Any deficiencies not covered by the PR will be automatically created as issues.
